### PR TITLE
Generate load tests from OpenAPI Spec + Load-test Uitlity API

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #8418 

## Description of Changes
- Creates a script to generate k6 load test from OpenAPI Spec
- Creates a private utility API for k6 load tests (protected via `LOAD_TESTING_AUTH_TOKEN`)
  - Only includes JWT creation for a random sample of users (for now).
- Modifies config merging to use lodash deep merging.
- Introduced a new pattern of defining default env var values for reusability.
- Makes `JWT_SECRET` env var required (non-default value) in production environments.
- Updated load-testing readme with creating test best practices (with references to k6 documentation).

## Test Plan
- Generate load tests from OpenAPI spec with `pnpm generate-load-tests` script in the root `package.json`
- Start the app locally to verify env var configuration

## Deployment Plan

1. Set `LOAD_TESTING_AUTH_TOKEN` in all Heroku environments **_before_** deployment.

## Other Considerations

-

### CODE REVIEW

**Comment by kurtisassad** [comment]
When running the script I get (From the openAPIGenerator docker container)

```
[main] ERROR io.swagger.v3.parser.util.RemoteUrl - unable to read
java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Unknown Source)
	at java.base/sun.nio.ch.Net.connect(Unknown Source)
	at java.base/sun.nio.ch.NioSocketImpl.connect(Unknown Source)
	at java.base/java.net.Socket.connect(Unknown Source)
	at java.base/java.net.Socket.connect(Unknown Source)
	at java.base/sun.net.NetworkClient.doConnect(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.openServer(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.openServer(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.<init>(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.New(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.New(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.connect(Unknown Source)
	at io.swagger.v3.parser.util.RemoteUrl.urlToString(RemoteUrl.java:151)
	at io.swagger.v3.parser.OpenAPIV3Parser.readContentFromLocation(OpenAPIV3Parser.java:290)
	at io.swagger.v3.parser.OpenAPIV3Parser.readLocation(OpenAPIV3Parser.java:95)
	at io.swagger.parser.OpenAPIParser.readLocation(OpenAPIParser.java:16)
	at org.openapitools.codegen.config.CodegenConfigurator.toContext(CodegenConfigurator.java:686)
	at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:744)
	at org.openapitools.codegen.cmd.Generate.execute(Generate.java:527)
	at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
```

Anyways maybe its best to see if others are getting this issue first, or if its just me

**Comment by kurtisassad** [comment]
Ok so it threw the error, but actually wrote the script correctly. This PR is absolute 🔥. Will approve this

**Review by kurtisassad — APPROVED** [review]

**Comment by timolegros** [comment]
> Ok so it threw the error, but actually wrote the script.js correctly. This PR is absolute 🔥. Will approve this

Yes it does throw some errors and most are to be expected. They outline in their docs that this is "best effort" generation - not all standards are transferrable to k6 and also some of our setup is non-standard (i.e. adapter pattern over tRPC). The main thing is that it gets the structure right and helps as a reference to build utility functions/tests.

File: libs/core/src/framework/types.ts [hunk 1]
```diff
@@ -175,6 +176,7 @@ export type CommandMetadata<
   readonly auth: CommandHandler<Input, Output>[];
   readonly body: CommandHandler<Input, Output>;
   readonly secure?: boolean;
+  readonly authStrategy?: AuthStrategies;
```
Comment (rbennettcw) [inline libs/core/src/framework/types.ts:179]: where's `authStrategy` being used within the framework?

Comment (timolegros) [reply libs/core/src/framework/types.ts:179]: It's used in the `authenticate` function -> `adapters/src/trpc/index.ts`. This function is used by both the command and query functions.

**Review by rbennettcw — APPROVED** [review]

**Pull request merged**